### ISSUE TITLE
Implement cell type assignment methods

### ISSUE THREAD

**Issue by nictru** [nf-core/scdownstream#20]
### Description of feature

The following methods look interesting:

```[tasklist]
### Tasks
- [ ] https://github.com/nf-core/scdownstream/issues/21
- [ ] https://github.com/nf-core/scdownstream/issues/22
- [ ] https://github.com/nf-core/scdownstream/issues/23
- [ ] https://github.com/nf-core/scdownstream/issues/24
```

**Comment by grst** [comment]
What's your take here on cell-type assignment by atlas-mapping? In my experience, these approaches work best, but are of course limited by the availability of such an atlas and are conceptually different than the methods mentioned above. 

The future will bring "foundational models" that are trained on all available data. [scsimilarity](https://genentech.github.io/scimilarity/)  goes in that direction -- it generates a low-dimensional embedding and cell-type labels in one go. 

The Theislab is working on [nicheformer](https://github.com/theislab/nicheformer). 

I would keep that in mind during the pipeline design that you could jump to cell-type labels + low dimensional embedding in one step.

**Comment by nictru** [comment]
This sounds very interesting.
Implementation-wise the pipeline can produce labels+embeddings in one step already. An example of this is how the [SCANVI](https://github.com/nf-core/scdownstream/blob/dev/modules/local/scvitools/scanvi/main.nf) process is implemented.

**Comment by vd4mmind** [comment]
If I may add, first iteration of cell type label assignment is better with some of the well benchmarked tool untill foundational models are fully operational and trustworthy. From cell type classification purposes, this first iteration label + embedding can still be reassigned and improved with human intervention if labels are a bit off, which will be case-specific (most likely depending on organ/tissue and disease type). So atlas level data will always serve as better ground truth training and for classification purposes with any of the transformer based or pure ML or Neural Network based models in future. So if such comes better, that can be implemented as a process.
